Content.Server/Ignitable/IgnitableSystem.cs

```diff
@@ -0,0 +1,185 @@
+        RaiseLocalEvent(args.Used, isHotEvent, false);
+
+        if (!isHotEvent.IsHot)
+            return;
+
+        Ignite(component);
+    }
+
+    private void Extinguish(IgnitableComponent component)
+    {
+        if(!component.CanRelight || component.DurationLeft <= 0)
+        {
+            SetState(component, SmokableState.Burnt);
+            return;
+        }
+
+        SetState(component, SmokableState.Unlit);
+    }
+
+    private void OnIsHotEvent(EntityUid uid, IgnitableComponent component, IsHotEvent args)
+    {
+        args.IsHot = component.CurrentState == SmokableState.Lit;
+    }
+
+    public void Ignite(IgnitableComponent component)
+    {
+        if (component.IsFirstLight)
+        {
+            IgnitableInit(component);
+        }
+
+        //If ShouldPlaySound is true then we've specified an ignition sound to play
+        if(component.ShouldPlaySound && component.IgniteSound.GetSound() != string.Empty)
+        {
+            SoundSystem.Play(
+                Filter.Pvs(component.Owner), component.IgniteSound.GetSound(), component.Owner,
+                AudioHelpers.WithVariation(0.125f).WithVolume(-0.125f));
+        }
+
+
+        //If this component has been set to 'Burnt' instead of Unlit then it is not possible to relight this component
+        if (component.CurrentState != SmokableState.Burnt)
+        {
+            SetState(component, SmokableState.Lit);
+            _litItems.Add(component);
+        }
+    }
+
+    private void SetState(IgnitableComponent component, SmokableState smokableState)
+    {
+        component.CurrentState = smokableState;
+
+        if (component.PointLightComponent != null)
+        {
+            component.PointLightComponent.Enabled = component.CurrentState == SmokableState.Lit;
+        }
+
+        if (EntityManager.TryGetComponent(component.Owner, out SharedItemComponent? item))
+        {
+            switch (component.CurrentState)
+            {
+                case SmokableState.Lit:
+                    item.EquippedPrefix = "lit";
+                    break;
+                default:
+                    item.EquippedPrefix = "unlit";
+                    break;
+            }
+        }
+
+        if (EntityManager.TryGetComponent(component.Owner, out AppearanceComponent? appearance))
+        {
+            //Set Ignitable AND Smoking Visuals as some ignitable things might have visualizers that make use of SmokingVisuals instead of IgnitableVisuals (Cigs)
+            appearance.SetData(IgnitableVisuals.SmokableState, component.CurrentState);
+            appearance.SetData(SmokingVisuals.Smoking, component.CurrentState);
+        }
+
+    }
+
+    private void IgnitableInit(IgnitableComponent component)
+    {
+        if(component.ShouldRandomize)
+        {
+            float numSecondsToChangeBy = _random.Next(-component.RandomizeMaxTime, component.RandomizeMaxTime);
+            component.Duration += numSecondsToChangeBy;
```

Comment:
Why can which go negative, numSecondsToChangeBy? Because we're either adding or removing duration. The duration being 10s is a mistaken commmit from me while I was testing.

If the default timer is 2000s (which it is) we're basically adding or removing UP to 120s from the timer, this just slightly changes the total burntime for a candle